### PR TITLE
Chemical washout

### PR DESCRIPTION
**Pull request by sbreiff**
A change in display_title for TreatmentAgent items so that if the treatment has a chemical and concentration of 0, it is labeled as "Chemical X washout" but otherwise it is labeled "Chemical treatment". Chemical treatments with no concentration indicated (e.g. None) are still labeled as treatment.

### CODE REVIEW

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/encoded/types/treatment.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5049103&bid=16625055)

File: src/encoded/types/treatment.py [hunk 1]
```diff
@@ -65,7 +65,8 @@ def display_title(self, request, treatment_type=None, chemical=None,
             conditions = " (" + "".join(d_t) + ")"
 
         if chemical:
-            dis_tit = chemical + " treatment" + conditions
+            word = ' treatment' if concentration != 0 else ' washout'
+            dis_tit = chemical + word + conditions
```
Comment (netsettler) [inline src/encoded/types/treatment.py]: I can't speak to whether "washout" is the right word here, but I'll assume that's a domain issue you have better understanding of than I do. It's clear what you're trying to do and functionally it seems fine.

While you're in there, though, I wouldn't mind seeing the variable `dis_tit` renamed to just `title` both for clarity and to reduce awkwardness of the name. I'd also probably rename `word` to `action` and change `chemical + word + conditions` to `chemical + " " + action + conditions` and change the `action` to not have a leading space in its value.

Optionally also change the setup for `conditions` (lines 54 to 65) to eliminate the `d_t` variable and just do something like this (untested, but I think the existing unit tests seem to be adequate--except there was no test of the weird case of no concentration or duration but still a temperature that leads to odd formatting in the existing code, hopefully fixed here):
```
        conditions = ""
        if concentration and concentration_units:
            conditions += str(concentration) + ' ' + concentration_units
        if duration and duration_units:
            if conditions:
                conditions += ', '
            conditions += str(duration) + duration_units[0]
        if temperature:
            if conditions:
                conditions += " "
            conditions += "at " + str(temperature) + "°C"
        if conditions:
            conditions = " (" + conditions + ")"
```

Comment (sbreiff) [reply src/encoded/types/treatment.py]: For the record, I'm not the one who originally created that variable title... but I'm happy to change it.

The comments about word vs action and placement of the space seem pretty trivial to me, but I can implement them if you think it makes the code read more clearly.

File: src/encoded/tests/test_types_treatment.py [hunk 2]
```diff
@@ -63,6 +63,20 @@ def test_calculated_chemical_treatment_display_title(testapp, drug_treatment):
     res = testapp.patch_json(drug_treatment['@id'], {'temperature': 3.5})
     assert res.json['@graph'][0]['display_title'] == 'Drug treatment (3.5 M, 3.5h at 3.5°C)'
 
+def test_calculated_chemical_treatment_display_title_temp_only(testapp, drug_treatment):
+    assert drug_treatment['display_title'] == 'Drug treatment'
+    res = testapp.patch_json(drug_treatment['@id'], {'temperature': 37})
+    assert res.json['@graph'][0]['display_title'] == 'Drug treatment (at 37°C)'
+
+
```
Comment (j1z0) [inline src/encoded/tests/test_types_treatment.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5049103)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/encoded/types/treatment.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5049103&bid=16625055)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/encoded/types/treatment.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5049103&bid=16625055)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/encoded/types/treatment.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5049103&bid=16625055)

**Pull request merged**